distribution/src/bin/elasticsearch

```diff
@@ -45,6 +50,15 @@ then
   fi
 fi
 
+if [[ $AUTO_CONFIG = true ]]; then
```

Comment:
We need to ensure that the arguments passed to `bin/elasticsearch` are not forwarded to the call to `AutoConfigInitialNode`.  As it is now if one runs `bin/elasticsearch -d` for instance, then `-d` is passed to `AutoConfigInitialNode` and it fails as it doesn't recognize this option

Reply:
Aaargh, yes, good point.
They need to be filtered, though; the `-E` options must be passed in to `AutoConfigInitialNode`. I forgot to consider that the tools is strict and doesn't ignore unknown options.

Reply:
I think we could also `parser.allowsUnrecognizedOptions();` in the AutoConfigInitialNode command